apollo-compiler/src/test/graphql/com/example/arguments_complex/TestQuery.kt

```diff
@@ -71,7 +71,7 @@ data class TestQuery(
   override fun parse(source: BufferedSource): Response<Data> = parse(source, DEFAULT)
 
   data class HeroWithReview(
-    val __typename: String,
+    val __typename: String = "Human",
```

Comment:
This is not supposed to be changed right? What about moving it out from the constructor and keep it as a field.

Reply:
Apparently this is still required for the cache: https://github.com/apollographql/apollo-android/issues/1936#issuecomment-578431702

I wonder if this is a technical limitation or if, for the same generated model, there's a use case for passing different values ? Maybe with inheritance ?

Reply:
It can be different. Right now the default value is taken from the definition of GraphQL schema type. But in case of interface the real value for `__typename` depends on the returned instance.
For example having this query:
```
query {
  hero {
    __typename
    name
  }
}
```

Object `hero` is defined as  `Character` in GraphQL schema. So default value will be `"Character"`. But at runtime it can be one of  possible types `Human` or `Droid` https://github.com/apollographql/apollo-android/blob/master/apollo-compiler/src/test/graphql/schema.json#L637.